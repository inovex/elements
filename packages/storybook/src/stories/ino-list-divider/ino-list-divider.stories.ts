import { html } from 'lit-html';

export default {
  title: 'Structure/ino-list-divider',
  component: 'ino-list-divider',
};

export const Playground = (args) => html`
  <ino-list>
    <ino-list-item text="Item 1"></ino-list-item>
    <ino-list-item text="Item 2"></ino-list-item>
    <ino-list-divider
      inset="${args.inset}"
      between-lists="${args.betweenLists}"
      padded="${args.padded}"
    ></ino-list-divider>
    <ino-list-item text="Item 3"></ino-list-item>
  </ino-list>
`;
Playground.args = {
  inset: false,
  betweenLists: false,
  padded: false
}

export const Inset = () => html`
  <ino-list>
    <ino-list-item text="Add to favorites">
      <ino-icon slot="leading" icon="favorite"></ino-icon>
    </ino-list-item>
    <ino-list-divider inset></ino-list-divider>
    <ino-list-item text="Star it!">
      <ino-icon slot="leading" icon="star"></ino-icon>
    </ino-list-item>
  </ino-list>
`;

export const BetweenLists = () => html`
  <ino-list>
    <ino-list-item text="List 1 - item 1"></ino-list-item>
    <ino-list-item text="List 1 - item 2"></ino-list-item>
    <ino-list-item text="List 1 - item 3"></ino-list-item>
  </ino-list>
  <ino-list-divider between-lists></ino-list-divider>
  <ino-list>
    <ino-list-item text="List 2 - item 1"></ino-list-item>
    <ino-list-item text="List 2 - item 2"></ino-list-item>
    <ino-list-item text="List 2 - item 3"></ino-list-item>
  </ino-list>
`;
