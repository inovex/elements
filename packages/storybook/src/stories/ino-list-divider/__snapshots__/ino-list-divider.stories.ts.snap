// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-list-divider BetweenLists smoke-test 1`] = `
<div id="root-inner">
  <ino-list>
    <ino-list-item text="List 1 - item 1">
    </ino-list-item>
    <ino-list-item text="List 1 - item 2">
    </ino-list-item>
    <ino-list-item text="List 1 - item 3">
    </ino-list-item>
  </ino-list>
  <ino-list-divider between-lists="true">
  </ino-list-divider>
  <ino-list>
    <ino-list-item text="List 2 - item 1">
    </ino-list-item>
    <ino-list-item text="List 2 - item 2">
    </ino-list-item>
    <ino-list-item text="List 2 - item 3">
    </ino-list-item>
  </ino-list>
</div>
`;

exports[`Structure/ino-list-divider Insert smoke-test 1`] = `
<div id="root-inner">
  <ino-list>
    <ino-list-item text="Add to favorites">
      <ino-icon slot="leading"
                icon="favorite"
      >
      </ino-icon>
    </ino-list-item>
    <ino-list-divider inset="true"
                      between-lists="false"
    >
    </ino-list-divider>
    <ino-list-item text="Star it!">
      <ino-icon slot="leading"
                icon="star"
      >
      </ino-icon>
    </ino-list-item>
  </ino-list>
</div>
`;

exports[`Structure/ino-list-divider Padded smoke-test 1`] = `
<div id="root-inner">
  <ino-list>
    <ino-list-item text="Item 2">
    </ino-list-item>
    <ino-list-divider inset="false"
                      between-lists="false"
                      padded="true"
    >
    </ino-list-divider>
    <ino-list-item text="Item 3">
    </ino-list-item>
  </ino-list>
</div>
`;

exports[`Structure/ino-list-divider Playground smoke-test 1`] = `
<div id="root-inner">
  <ino-list>
    <ino-list-item text="Item 2">
    </ino-list-item>
    <ino-list-divider inset="false"
                      between-lists="false"
                      padded="false"
    >
    </ino-list-divider>
    <ino-list-item text="Item 3">
    </ino-list-item>
  </ino-list>
</div>
`;
